features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx

```diff
@@ -57,188 +272,1561 @@ const useStyles = makeStyles((theme) => ({
  */
 export default function Environments() {
     const classes = useStyles();
+    const intl = useIntl();
     const { api, updateAPI } = useContext(APIContext);
     const { settings } = useAppContext();
-    const [gatewayEnvironments, setGatewayEnvironments] = useState([...api.gatewayEnvironments]);
+    let revisionCount;
+    if (Configurations.app.revisionCount) {
+        revisionCount = Configurations.app.revisionCount;
+    } else {
+        revisionCount = 5;
+    }
     const [selectedMgLabel, setSelectedMgLabel] = useState([...api.labels]);
-    const [isUpdating, setUpdating] = useState(false);
     const [selectedDeployments, setSelectedDeployments] = useState([...api.deploymentEnvironments]);
-
     const restApi = new API();
     const [allDeployments, setAllDeployments] = useState([]);
+    const [allRevisions, setRevisions] = useState(null);
+    const [allEnvRevision, setEnvRevision] = useState(null);
+    const [selectedRevision, setRevision] = useState(null);
+    const [extraRevisionToDelete, setExtraRevisionToDelete] = useState(null);
+    const [description, setDescription] = useState('');
+    const [mgLabels, setMgLabels] = useState([]);
+    const [SelectedEnvironment, setSelectedEnvironment] = useState([]);
+    const [open, setOpen] = useState(false);
+    const [confirmDeleteOpen, setConfirmDeleteOpen] = useState(false);
+    const [revisionToDelete, setRevisionToDelete] = useState([]);
+    const [confirmRestoreOpen, setConfirmRestoreOpen] = useState(false);
+    const [revisionToRestore, setRevisionToRestore] = useState([]);
+    const [openDeployPopup, setOpenDeployPopup] = useState(false);
+    const [lastRevisionCount, setLastRevisionCount] = useState(0);
+
+    const extractLastRevisionNumber = (list, lastRev) => {
+        if (lastRev !== null) {
+            setLastRevisionCount(lastRev);
+        } else if (list[list.length - 1]) {
+            const lastRevisionName = list[list.length - 1].displayName;
+            const splitList = lastRevisionName.split(' ');
+            setLastRevisionCount(splitList[1]);
+        } else {
+            setLastRevisionCount(0);
+        }
+    };
+
     useEffect(() => {
         restApi.getDeployments()
             .then((result) => {
                 setAllDeployments(result.body.list);
             });
+        restApi.microgatewayLabelsGet()
+            .then((result) => {
+                setMgLabels(result.body.list);
+            });
+        restApi.getRevisions(api.id).then((result) => {
+            setRevisions(result.body.list);
+            setLastRevisionCount(result.body.count);
```

Comment:
Here setting `setLastRevisionCount` is redundant since the same state is updated inside the `extractLastRevisionNumber` as well. And also I think this last revision number should have derived from the revisions list rather than keeping that number in a state